### ISSUE TITLE
New Block: Icon Box

### ISSUE THREAD

**Issue by Soare-Robert-Daniel** [Codeinwp/otter-blocks#610]
A new block that is similar to Icon List but with more specific customization and additional features.

Usage case:
![target](https://user-images.githubusercontent.com/52494172/138232406-20c2b377-2a5c-4743-8b0c-fc1083e536ac.png)

Issue Ref: #530

**Comment by HardeepAsrani** [comment]
@mghenciu Do you want it as a block or should we do it as a Variant just like we are doing with the new Pricing Block?

**Comment by HardeepAsrani** [comment]
@mghenciu Sure. Can you list some of the options you'd want to see in the block and we can add it to the pipeline? Thanks!

And if you can have an HTML version somewhere, that will be great too!

**Comment by JohnPixle** [comment]
Just some thoughts on this @HardeepAsrani, @Soare-Robert-Daniel  and @mghenciu:

Currently we have the service block, which essentially can behave the same as the suggested icon box.

Regarding the icon positioning that @mghenciu mentioned, I agree, it should be easy to switch (icon on the left or right side, or at the top.
This is what the "icon position" controls are about in this screenshot. I believe this positioning can be achieved with CSS.
![Screenshot 2022-03-22 at 16 22 45](https://user-images.githubusercontent.com/4193672/159504148-0a8aa3f2-2f44-473c-8a11-22ef06ecfe00.png)

Additionally, the user is not limited to using an icon, he can add an image if he wants. 
SO this block can behave an an image box OR as an icon box.

Now, the advantage in Mihai's example from Elementor is obvious. Since the widget is "fixed", one can have more control over all the elements of the block at once. For example, change the text alignment, icon style etc.

This is not possible in the case of "open" blocks like our Service block, testimonial etc. Let's discuss further.

**Comment by mghenciu** [comment]
> @mghenciu Sure. Can you list some of the options you'd want to see in the block and we can add it to the pipeline? Thanks!
> 
> And if you can have an HTML version somewhere, that will be great too!

- [ ] First thing, which is not necessarily setting related, is the vertical alignment. It should work well with or without the paragraph text, and it should be centered vertically. Here's an example with Icon Box description visible, and without it, where you can see that the vertical alignment behaves depending if the paragraph is present or not:
<img width="401" alt="image" src="https://user-images.githubusercontent.com/52494172/160380670-cd2864d7-e819-4cf4-918f-44bdd2f7e25d.png">
<img width="1228" alt="image" src="https://user-images.githubusercontent.com/52494172/160380904-190d529e-91d2-4113-8d2d-da5e70552de1.png">

### Options
- [ ] Icon position: Left, Top, Right
- [ ] Link, which will be on the Icon Box Title and on the icon
- [ ] Option to select the HTML tag for the Title
- [ ] Styles: Default, Fill, Border
- [ ] by default the Icon container will be a square, but with border radius it can be transformed into a circle.
- [ ] option to upload custom Image/SVG
- [ ] Icon controls (size, padding, etc). 
- [ ] Also here, we need a spacing control for the distance between the Icon and Text(content). And spacing between the Title & Description.
- [ ] Alignment options for the content (title + description). One control for both of them: Left, Center, Right. 
Based on device.

- [ ] On Mobile, even if the Icon box is set to Left/Right, the icon will go on top. Basically the usual responsive behavior when content goes in one column. But making sure the Icon remains on Top.

**Comment by JohnPixle** [comment]
Hi all
Regarding this, we already have a Service block, which visually is similar to this, right? 
But as far as I understand, this new icon box will be dynamic and come with more flexible settings. I agree that this is going to be useful.

Some ideas to expand on Mihai's input:
Perhaps we should not restrict it to icon box in the title. What about having the option to use an icon OR an image? At Elementor they have the Icon Box, and the Image Box. Perhaps we could combine some usability into a single powerful block.

So if we rename it into something as "Content box" or "Feature box" or something along these lines...

In the screenshot below are all the possible variations for images and icons, based on Mihai's input on settings.
I already had those in Figma ready btw.

I also include an idea for a "Cover" variant which can be PRO.
 
![feature box draft](https://user-images.githubusercontent.com/4193672/161591408-239d43c2-8514-4f45-a290-f56cee0ee553.png)

Let me know your thoughts, happy to start a mockup for the inspector sidebar for this block, so we can have a common reference.

**Comment by HardeepAsrani** [comment]
@JohnPixle I like the ideas. I also see our lack of foresight here in the past year with the Flip block. I think Flip Block feature could've also been part of a Feature Block. What do you think?

**Comment by mghenciu** [comment]
> @JohnPixle I like the ideas. I also see our lack of foresight here in the past year with the Flip block. I think Flip Block feature could've also been part of a Feature Block. What do you think?

Even though the Flip block & Feature Block (this one) have some things in common and can share some settings between them, I think **there's one thing that makes them very different**. And that's how the content is added. 

Basically with the Feature Block - users will have the option to add an Icon, Title, Description, and Button if they want.
While with the Flip Card - they can add whatever they want inside the card (social icons, progress bars, etc).

**Comment by HardeepAsrani** [comment]
@mghenciu That's true but only one side of the Flip allows free movement of blocks, while the front is pre-defined fields.

@JohnPixle You can start with the designing.

**Comment by JohnPixle** [comment]
I am leaning towards keeping the two blocks separate. I think they serve different purposes, and having them all integrated into a single block will cost the user some flexibility. 

The flip box ideally should have been "open" or "closed" on both sides, currently it is a bit confusing that the two sides behave in a different way. However we can't change it now assume.

I will start wrapping up the UI for the feature block and follow up here.

**Comment by JohnPixle** [comment]
Just a heads-up that I started the design for the [Feature Block controls](https://www.figma.com/file/LRrhO1ovZnXFibXVsezR8a/Otter?node-id=689%3A2574) in Figma. @HardeepAsrani @mghenciu There might be some things that need to be changed in this mockup, please share your insights and let me know your thoughts overall. 

Initially, Hardeep, let me know if the inspector's sidebar layout can be achieved or I need to make changes to the mockup?

Thanks!

**Comment by HardeepAsrani** [comment]
@JohnPixle Maybe we don't need "Add Button" option? We can just have the field (in some tab or a tab of its own) with an empty URL and label. We will only how it in the front (and if the block isn't selected in the editor) if the user has defined either link or label.

That will be better than adding an extra attribute to disable/enable button control.

**Comment by JohnPixle** [comment]
@mghenciu Yes, I definitely agree there needs to be some consistency across the blocks. 

> Those would probably go under the Dimensions tab (after seeing your prototype, but before we weren't sure), and we need to keep in mind things like this, when refreshing our blocks -> consistency.

I get your point, but here is the case: From what I see currently in core, they indeed use the dimensions panel to expose controls for spacing, but **these controls refer to the entire block**. So for example, In the form case, under Dimensions panel we would add spacing controls (padding / margin) that would apply on the the form block entirely, and not individual nested fields. It would be better to have a different panel for managing micro-spacing on these components.

I will take a look around on the form ticket and follow up if needed.

> We can just have the field (in some tab or a tab of its own) with an empty URL and label. We will only how it in the front (and if the block isn't selected in the editor) if the user has defined either link or label.

Absolutely, great idea. Will update the Figma.

**Comment by HardeepAsrani** [comment]
@JohnPixle This was talked about in the last meeting and it was discussed that it might not be possible to cover all cases of an Icon Box in Patterns and it will be hard for people to make minor changes, so we had thought of starting with Patterns at first but later making an Icon Block. This way we will also have more ideas on what things we need for a block.

**Comment by JohnPixle** [comment]
Sounds good to me, thanks for the input!

**Comment by mghenciu** [comment]
Sorry for the late reply, @JohnPixle .
The idea and concerns I raised in the meeting were related to how would this work at this point, and here a 2 points:

**1. Building an Icon Box with the Section Block**
Right now building an Icon Box will require using the Section block, and at this point this is quite challenging especially in terms of spacing and Responsiveness, or when having a lot of icons boxes in a Layout (for example 4 columns, each of them having an Icon Box).

**2. Quickly switching between Styles**
The most useful thing in Elementor (and probably you are aware of this), is the easiness to switch between styles and configurations, without having to redo that section from scratch. I attached a video from Elementor, where it's visible how easy is to switch between Right, Top, Left icon position, same thing for aligning the content - and all those settings being device dependent.

https://user-images.githubusercontent.com/52494172/190619592-a4124e09-a395-4598-830b-a7429c3b2770.mov




So that's why, imo, having the Icon/Image box - makes more sense in a long run.

**Comment by JohnPixle** [comment]
I definitely agree @mghenciu . It would make sense to give it a shot after we get the sidebar reconfiguration and stand on a more solid ground with the UIX. 
Thanks both for the input.
